### PR TITLE
Add range status in filters

### PR DESCRIPTION
**Pull request by cedvan**
## Need configure range status in filters logs

Add **multiples formats** to configure status in filters : 

- `[401, 404]` : Add status `401` and `404`
- `[^456]` : Add status hundred greater than 456 (`456, 457, 458, ..., 499`)
- `[4*]` : Add status hundred (`400, 401, 402, ..., 499`)
- `[41*]` : Add status decade (`410, 411, 412, ..., 419`)
- `[425-440]` : Add range status (`425, 426, 427, ..., 440`)
- `[4*, !^453]` : Add status `400, 401, 402, ..., 452`
- `[4*, !415-418]` : Add status hundred 400 (`400, 401, 402, ..., 499`) but exclude range `415` to `418`
- `[425-440, !427, !430]` : Add range status (`425, 426, 427, ..., 440`) but exclude `427` and `430`
- `[2*, 301, 400-404, !403]` : Add status `200, 201, ..., 299, 301, 400, 401, 402, 404`

Instead of add can be use `!` to exclude status

Order is important :
- `[4*, !^453]` : Add status `400, 401, 402, ..., 452`
- `[!^453, ~450]` : Add status `400, 401, 402, ..., 499`

### CODE REVIEW

File: src/M6Web/Bundle/LogBridgeBundle/Tests/Resources/config/config.yml [hunk 1]
```diff
@@ -26,22 +26,27 @@ filters:
         method: ['PUT']
         status: ~
 
-    get_clip_500:
+    get_clip_500_all_hundred:
```
Comment (mikaelrandy) [inline src/M6Web/Bundle/LogBridgeBundle/Tests/Resources/config/config.yml]: Je n'aime pas ce nomage.
Je ne veux pas avoir à modifier le nom parce que je change la liste des status.

Pour moi, on remonte les clips en erreur : 'get_clip_error'

Comment (mikaelrandy) [inline src/M6Web/Bundle/LogBridgeBundle/Tests/Resources/config/config.yml]: J'ai rien dit, je n'avais pas vu que c'était des tests

**Comment by omansour** [comment]
sounds good. mais ça serait pas mal de faire un TU sur parseStatus non ?

**Comment by cedvan** [comment]
@omansour Yes, I'll make ;)

File: src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php [hunk 2]
```diff
@@ -417,4 +418,98 @@ private function compileFilterStatus($prefix, $filter)
...
+                    $statusList = array_merge($statusList, $addStatusList);
+                    break;
+
+                // Status range hundred or decade (examples: 4* => [400, 401, ..., 499] or 45* => [450, 451, ..., 459])
+                case (preg_match('/^[\d]{1,2}\*(![\d]{3})*$/', $value) ? true : false) :
+                    $filterStatus = explode('*', $value)[0];
+
+                    if (strlen($filterStatus) == 2) {
+                        $rangeInternval = 10;
+                    } else {
```
Comment (Fabex) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php]: Comme d'hab je suis chiant mais ce else sert à rien ;)

Comment (cedvan) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php]: Yes , I remove ;)

**Comment by Fabex** [comment]
C'est bien; mais perso pour le gros bout de code bien fat du case dans parseStatus, j'aurais plutot fais un système de voter pour chaque régle. ca deviendrais tout de suite plus lisible et plus facilement étendable

**Comment by mikaelrandy** [comment]
When reading, i had the same feeling as @Fabex : the use/case is a fat piece of code.
I like the "voters" idea !

**Comment by cedvan** [comment]
@Fabex @mikaelrandy , yes voters is a great idea ! I'll make ;)

File: src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php (hunk 2 above)
Comment (omansour) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php]: mais qu'il est chiant ;)

File: src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php [hunk 3]
```diff
@@ -405,11 +405,12 @@ private function compileFilter(Filter $filter)
     private function compileFilterStatus($prefix, $filter)
     {
         $compiled = [];
+        $filterStatusList = $filter->getStatus();
 
-        if (is_null($filter->getStatus())) {
+        if (is_null($filterStatusList)) {
             $compiled[] = sprintf('%s.all', $prefix);
         } else {
```
Comment (Fabex) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php]: à et au fait y'a ce else aussi ;)

Comment (cedvan) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Dumper/PhpMatcherDumper.php]: done

**Comment by cedvan** [comment]
@fabdsp @mikaelrandy done

File: src/M6Web/Bundle/LogBridgeBundle/Matcher/Status/Type/AbstractType.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+     * @var string
+     */
+    protected $regex = null;
+
+    /**
+     * Type Constructor
+     */
+    public function __construct()
+    {
+        if ($this->regex === null) {
```
Comment (fabdsp) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Status/Type/AbstractType.php]: I would prefer an abstract method getRegex (or Maybe getPattern ?) to be defined in the concrete class - so no if, no exception etc.

Comment (cedvan) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Status/Type/AbstractType.php]: Ops miss, I'll fix ;)

Comment (cedvan) [inline src/M6Web/Bundle/LogBridgeBundle/Matcher/Status/Type/AbstractType.php]: done

**Comment by valentin-claras** [comment]
:+1:

**Comment by mikaelrandy** [comment]
I don't understand the difference betweet `[~450]` and `[~4*]`
I think you  originally want to reuse composer syntax, but `450` is not `4.5.0` and i think it could become a real concern for understanding.

**Comment by cedvan** [comment]
`[~4*]` isn't possible. But `[~450]` and `[4*]` hasn't differences, yes. You want me to remove  `TypeDecade` (`[~450]`) ?

**Comment by mikaelrandy** [comment]
As you said, you want to reuse composer syntax, but How do you handle status 4010 ? As `4.0.10`, `40.0.1`, `4.01.0` ?

I think the `[4*]` syntax is clear and perfect and yes, i think we should remove `[~450]` syntax

**Comment by cedvan** [comment]
@mikaelrandy, Yes, you're right, I'll remove syntax `[~450]`

**Comment by cedvan** [comment]
@mikaelrandy `[~450] syntax is removed`

**Comment by cedvan** [comment]
up ?

**Comment by fabdsp** [comment]
:+1:

**Comment by mikaelrandy** [comment]
:+1:

**Comment by Fabex** [comment]
:+1:

**Pull request merged**